### DISCUSSION TITLE
Prevent invitation emails from ending up as spam

### DISCUSSION THREAD

**Discussion by Striffly** [plausible/analytics#3225] [Self-Hosted Support]
When a user is invited to Plausible, the e-mail they receive is likely to end up in the spam folder, as the "score" associated with this e-mail is low.

![image](https://github.com/plausible/analytics/assets/13284087/c89a1e67-0752-4a9d-8763-39697c1833ad)
(test done with https://www.mail-tester.com/)

It would be nice if these recommendations were taken into account

**Comment by ruslandoga** [comment]
👋 @Striffly

Thank you for the recommendations!

I wonder how low the final score is. The website suggests that "a score below -5 is considered spam," but the numbers on your screenshot don't add up to that.

Also, are you using the default SMTP adapter? An easy way to up the score would be to switch to a hosted mail service like Mailgun, etc. AFAIK, the SMTP adapter's goal is mostly ease of use, not avoiding the spam folder.

**Reply by ruslandoga** [reply]
The sent email does seem to contain both the message-id and date headers (check the "Source" section), not sure why they are shown as missing by spamassasin.

**Comment by dimejo** [comment]
@ruslandoga:
> The sent email does seem to contain both the message-id and date headers (check the "Source" section), not sure why they are shown as missing by spamassasin.

How exactly do you send emails? I'm setting `SMTP_*` env variables for the docker container and all emails I checked neither contain `Date` nor `Message-Id` headers.

```
Return-Path: <sender@example.net>
Delivered-To: receiver@example.org
Received: from mail.ud09.udmedia.de ([127.0.0.1])
	by mail.ud09.udmedia.de with LMTP
	id KDJvNoUFQmVyQD8ALK0PDQ
	(envelope-from <sender@example.net>)
	for <receiver@example.org>; Wed, 01 Nov 2023 09:00:05 +0100
Delivered-To: receiver@example.org
Received: (qmail 4145525 invoked from network); 1 Nov 2023 09:00:05 +0100
Received: by mail.ud09.udmedia.de with ESMTPSA (TLS_AES_256_GCM_SHA384 encrypted, authenticated); 1 Nov 2023 09:00:05 +0100
X-UD-Smtp-Session: receiver@b/PDqBIJIbsxDRHI
Subject: Test
From: <sender@example.net>
To: receiver@example.org
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="----=_Part_23302832_498096969.3694877952"

------=_Part_23302832_498096969.3694877952
Content-Type: multipart/alternative; boundary="----=_Part_3852777623_3070413339.1951176649"

------=_Part_3852777623_3070413339.1951176649
Content-Type: text/html;charset=UTF-8
Content-Transfer-Encoding: base64

[...]
```

**Reply by ruslandoga** [reply]
I don't remember anymore, but I was most likely using the default setup. Which is sending the mail through the [`mail`](https://github.com/plausible/hosting/blob/bb6decee4d33ccf84eb235b6053443a01498db53/docker-compose.yml#L3-L5) container.

**Reply by madanyang** [reply]
I can confirm the missing message-id and Date headers in the sent messages. Plausible hosted in the cloud and it is configured to use mailserver on premises via the port 587 authentication. Amavis is marking the mails sent by plausible with bad-header warning.

**Reply by ruslandoga** [reply]
@madanyang 👋

What's a mailserver? If it's a relay, then it should add the proper headers if they are missing before sending the mail to the end users. That's what [`mail`](https://github.com/plausible/hosting/blob/bb6decee4d33ccf84eb235b6053443a01498db53/docker-compose.yml#L3-L5) does, for example.

**Reply by madanyang** [reply]
It is not a relay. Plausible authenticates to the mail server and sends the mail. In your example bytemark/smtp is generating them. In my case plausible is connecting to the mail server directly. Date should be the responsiblity of mail client not the server. In this case plausible acting as a client does not create a Date field.

**Reply by ruslandoga** [reply]
[The current library](https://github.com/gen-smtp/gen_smtp) that's used for SMTP is supposed to only be used with relays. The next release might add [another client](https://github.com/ruslandoga/mua) that would allow direct sending.

**Reply by dimejo** [reply]
Quoting from [RFC 5322](https://www.rfc-editor.org/rfc/rfc5322.html#section-3.6):

> The only required header fields are the origination date field and the originator address field(s).  All other header fields are syntactically optional. More information is contained in the table following this definition.


Quoting from [RFC 5322](https://www.rfc-editor.org/rfc/rfc5322.html#section-3.6.1):

> 
> The origination date field consists of the field name "Date" followed by a date-time specification.
> 
> orig-date       =   "Date:" date-time CRLF
> 
> The origination date specifies the date and time at which the creator of the message indicated that the message was complete and ready to enter the mail delivery system. For instance, this might be the time that a user pushes the "send" or "submit" button in an application program. In any case, it is specifically not intended to convey the time that the message is actually transported, but rather the time at which the human or other creator of the message has put the message into its final form, ready for transport. (For example, a portable computer user who is not connected to a network might queue a message for delivery. The origination date is intended to contain the date and time that the user queued the message, not the time when the user connected to the network to send the message.)

To me this sounds like the `Date:` header needs to be created when the message is created by the client (in this case the gen_smtp library).

The reason why this is not an issue (at least to my understanding) when using the `bytemark/smtp` image is because the image is using Exim under the hood. By default [Exim automatically adds the `Date:` header](https://www.exim.org/exim-html-current/doc/html/spec_html/ch-message_processing.html#SECID223) when it is missing. Unfortunately some other MTAs like Postfix or qmail do not (for various reasons) add missing headers by default.

**Comment by sujiba** [comment]
Hello,

I have the same issue with mails sent from plausible. They are marked as spam because of a missing-messageid and missing-date. 

Has someone found a solution in the meantime?

Plausible Version 3.0.1

Plausible mail settings in .env:

- MAILER_EMAIL=stats@example.com
- MAILER_NAME='stats'
- SMTP_HOST_ADDR=mx.example.com
- SMTP_HOST_PORT=25
- SMTP_HOST_SSL_ENABLED=false

Kind regards
sujiba

**Reply by dimejo** [reply]
This should be fixed by https://github.com/plausible/analytics/pull/4474 and a quick test on v3.0 shows `Date` and `Message-Id` headers for me.

**Reply by sujiba** [reply]
Hi,

thank you for your fast response. You're right, the message-id and date are present. As far as I can tell, the problem is that the message-id has the wrong format. 

Instead of: 

- Message-Id: 0210d0sadak2e2

it should be something like this: 

- Message-Id: 0210d0sadak2e2@example.com

See also [rfc2822](https://www.ietf.org/rfc/rfc2822.txt ) Appendix A. Example messages (Page 41).

Maybe I'm wrong.

**Reply by dimejo** [reply]
> it should be something like this:
> 
>     Message-Id: 0210d0sadak2e2@example.com

Correct, but `<>` brackets are also required.

@ruslandoga:
Looks like this was fixed in [bamboo_mua v0.2.3](https://github.com/ruslandoga/bamboo_mua/blob/master/CHANGELOG.md), but Plausible is still using v0.2.2.

**Reply by sujiba** [reply]
Thanks again for pointing me in the right direction. Than I'm going to wait for the next release.

**Reply by sujiba** [reply]
Hi @ruslandoga and @apata , 

will bamboo_mua also be updated with the release 3.1.0? The taged release shows [0.2.2](https://github.com/plausible/analytics/blob/v3.1.0-rc.1/mix.lock)

**Reply by apata** [reply]
@sujiba Thanks for the ping! We will update it by the next RC or for the actual release.

**Reply by sujiba** [reply]
Thank you. 🙂
